cli/mtv_api_tests/run.py

```
@@ -0,0 +1,315 @@
+        console.print(f"[red]Error: {providers_path} not found. Run 'mtv-api-tests generate' first.[/red]")
+        raise typer.Exit(code=1)
+
+    resolved_category = select_category(category)
+
+    test_filter = prompt_test_filter(test_filter)
+
+    try:
+        providers = json.loads(providers_path.read_text())
+    except json.JSONDecodeError as exc:
+        console.print(f"[red]Error: {providers_path} contains invalid JSON: {exc}[/red]")
+        raise typer.Exit(code=1)
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)